moped-editor/src/components/ActivityMetrics.js

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from "react";
+import { useMutation } from "@apollo/client";
+import { LAST_SEEN_MUTATION } from "src/queries/staff";
+import { useUser } from "src/auth/user";
+/**
+ * This wrapper calls the set_last_seen endpoint, which updates the user's
+ * last_seen_date timestamp based on their session ID. this action happens
+ * once each time this component mounts
+ */
+export default function ActivityMetrics({ children }) {
+  const { user } = useUser();
+  const [setLastSeen] = useMutation(LAST_SEEN_MUTATION);
+  useEffect(() => {
+    if (!user) {
+      // this event can fire when the user is not logged in
+      return;
+    }
+    try {
+      setLastSeen();
+    } catch (error) {
+      console.error("Failed to set the last seen date for the current user.");
+    }
```

Comment:
an error would happen if Hasura bombed, or if we have user de-sync issue where where the user's ID is not actually in the DB.